recipes/nsqadmin.rb

```diff
@@ -13,19 +13,52 @@
 nsq_release = "nsq-#{node['nsq']['version']}-#{node['nsq']['go_version']}"
 
 if node['nsq']['setup_services']
-  template '/etc/init/nsqadmin.conf' do
-    action :create
-    source 'upstart.nsqadmin.conf.erb'
-    mode '0644'
-    # need to stop/start in order to reload config
-    if node['nsq']['reload_services']
-      notifies :stop, 'service[nsqadmin]', :immediately
-      notifies :start, 'service[nsqadmin]', :immediately
+
+  if node.platform?('ubuntu') && node['platform_version'].to_f >= 18.04
+
+    template '/etc/systemd/system/nsqadmin.service' do
+      action :create
+      source 'systemd.nsqadmin.conf.erb'
+      mode '0644'
+      notifies :run, 'systemd_unit[nsqadmin.service]', :delayed
+      # need to stop/start in order to reload config
+      if node['nsq']['reload_services']
+        notifies :stop, 'service[nsqadmin]', :immediately
+        notifies :start, 'service[nsqadmin]', :immediately
+      end
+    end
+
+    template "#{node['chef-nsq']['script_dir']}/nsqadmin-start.sh" do
```

Comment:
this should be moved out of the `if systemd` block, since it's used in both upstart and systemd

Reply:
This shell script is only used by the systemd service file. The upstart config for nsqadmin has the script in the [file](https://github.com/sproutsocial/chef-nsq/blob/main/templates/default/upstart.nsqadmin.conf.erb#L10-L28)

Reply:
ohhh okay